### PR TITLE
FindRepositoriesBlobs endpoint

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java [hunk 1]
```diff
@@ -1062,4 +1063,138 @@ private String getStringFromLocationList(List<String> locationList) {
...
+
+      Set<String> rootShaList = (Set<String>) resultSetMap.get("result");
+      Long totalRecords = (Long) resultSetMap.get("count");
+
+      Set<BlobExpanded> blobExpandedSet = new LinkedHashSet<>();
+      for (String rootSha : rootShaList) {
+        Map<String, BlobExpanded> blobExpandedMap =
+            getCommitBlobMap(session, rootSha, request.getLocationPrefixList());
+        blobExpandedSet.addAll(blobExpandedMap.values());
+      }
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: this is called in for loop for each commit root_hash so if there will be a multiple repo_ids in a request then this loop iterate for all the commit of the repos.

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java [hunk 2]
```diff
@@ -1062,4 +1063,145 @@ private String getStringFromLocationList(List<String> locationList) {
...
+    StringBuilder joinClause = new StringBuilder();
+    joinClause
+        .append(" INNER JOIN ")
+        .append(InternalFolderElementEntity.class.getSimpleName())
+        .append(" folderElm ")
+        .append(" ON ");
+    joinClause.append("folderElm.folder_hash = ").append(alias).append(".rootSha ");
+
+    StringBuilder whereClause = new StringBuilder();
+    boolean isNeedOperator = false;
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: similar variable renaming feedback as other PR

Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: Changed

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java [hunk 3]
```diff
@@ -1062,4 +1111,149 @@ private String getStringFromLocationList(List<String> locationList) {
...
+    }
+    List<String> resultSet = query.list();
+
+    Map<String, Object> responseMap = new HashMap<>();
+    responseMap.put("result", new HashSet<>(resultSet));
+    responseMap.put("count", countQuery.uniqueResult());
+    return responseMap;
+  }
+
+  private Boolean blobTypeExistsRequest(List<BlobType> blobTypeList, Blob.ContentCase contentCase)
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]:
```suggestion
  private Boolean blobTypeExistsInList(List<BlobType> blobTypeList, Blob.ContentCase contentCase)
```

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java [hunk 4]
```diff
@@ -1062,4 +1111,149 @@ private String getStringFromLocationList(List<String> locationList) {
...
+    joinClause
+        .append(" INNER JOIN ")
+        .append(InternalFolderElementEntity.class.getSimpleName())
+        .append(" folderElm ")
+        .append(" ON ");
+    joinClause.append("folderElm.folder_hash = ").append(alias).append(".rootSha ");
+
+    StringBuilder whereClause = new StringBuilder();
+    boolean appendAND = false;
+    if (!request.getRepoIdsList().isEmpty()) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java:1136]: use the AND joiner logic which you did on a different PR

**Review by ravishetye — COMMENTED** [review]
minor changes needed , but can merge post those changes

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java]: Changed

File: backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java (hunk 4 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java:1136]: Done

**Review by ravishetye — APPROVED** [review]

**Pull request merged**